src/lsp_data.rs

```diff
@@ -8,77 +8,80 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use std::collections::HashMap;
 use std::fmt::Debug;
 use std::path::{Path, PathBuf};
 
+use std::convert::TryFrom;
+
 use analysis::Span;
+use analysis::raw;
 use hyper::Url;
-use serde::{Serialize, Serializer};
+use serde::{Serialize};
+
+
+pub use ls_types::*;
 
 macro_rules! impl_file_name {
     ($ty_name: ty) => {
         impl $ty_name {
             pub fn file_name(&self) -> PathBuf {
-                let uri = Url::parse(&self.uri).unwrap();
-                uri.to_file_path().unwrap()
+                uri_string_to_file_name(&self.uri)
             }
         }
     }
 }
 
-/// Position in a text document expressed as zero-based line and character offset.
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct Position {
-    pub line: usize,
-    pub character: usize,
+pub fn uri_string_to_file_name(uri: &str) -> PathBuf {
+    let uri = Url::parse(&uri).unwrap();
+    uri.to_file_path().unwrap()
+}
+
+pub fn from_usize(pos: usize) -> u64 {
+	TryFrom::try_from(pos).unwrap() // XXX: Should we do error handling or assume it's ok?
+}
+
+pub fn to_usize(pos: u64) -> usize {
+	TryFrom::try_from(pos).unwrap() // FIXME: for this one we definitely need to add error checking
 }
```

Comment:
I would also just use a cast for this tbh, at least for ids, it seems very unlikely that we'll overflow a u32 and even using 32 bit machines for dev is rare nowadays.